plugins/link: shortname is empty if not linked to a game

This happens because there isn't a proper check that sets the shortname to "Link" in case the plugin is not linked to a game, as we use **wcsncpy**, that does not return an error.
On the Windows version of Link we use **wcscpy_s** with the proper check for the returned error, but it isn't available on Linux.
I searched for an alternative and found **[wcpncpy](http://linux.die.net/man/3/wcpncpy)**, that returns a pointer to the last character written; however, it's not a good solution, because the last character can be 0, as the string is null terminated.

Warning when compiling and the shortname is still empty.
```
link-posix.cpp:49:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (int i = 0; i < len; i++) {
                       ^
```

--- a/plugins/link/link-posix.cpp
+++ b/plugins/link/link-posix.cpp
@@ -18,7 +18,7 @@
 #include <fcntl.h>
 #include <time.h>
 
-static std::wstring wsPluginName;
+static std::wstring wsPluginName(L"Link");
 static std::wstring wsDescription;
 static char memname[256];
 

